### DISCUSSION TITLE
Troubleshooting access on BigQuery

### DISCUSSION THREAD

**Discussion by rajsinghUSA** [MIT-LCP/mimic-code#1453] [MIMIC-General] [answered]
In following the instructions [here](https://mimic.mit.edu/docs/gettingstarted/cloud/bigquery/), after clicking either of the the "+ ADD DATA" buttons shown here:
![image](https://user-images.githubusercontent.com/17286177/207450098-9d239970-fa6f-4821-a7ab-1eb31443ad16.png)

After clicking the "+ ADD DATA" button, a panel is displayed (shown in the following screenshot).
However, you can see there is no "Pin a project" button/option

![image](https://user-images.githubusercontent.com/17286177/207450435-ffade777-f54e-45c6-8750-c5156620aeb4.png)

Am I missing some step? Or has the flow in BigQuery changed?

**Comment by briangow** [comment]
Yes, the BigQuery interface has changed. Please select "Search for and star a project" from your second screen shot. This will do the same thing as "pin" previously did.

**Comment by rajsinghUSA** [comment]
Thanks @briangow 
I still do not see any results for "physionet-data" to Star:

![image](https://user-images.githubusercontent.com/17286177/207622175-c2ee83e7-86b9-41b1-a83d-9993153c228d.png)

**Reply by briangow** [reply]
Have you already completed the [training](https://physionet.org/settings/training/) and [credentialing](https://physionet.org/settings/credentialing/) requirements and followed all of the instructions outlined here: https://mimic.mit.edu/docs/gettingstarted/cloud/ ?

**Reply by rajsinghUSA** [reply]
Yes, training and credentialing has been completed. In fact, I was able to access the MIMIC data on AWS, just not on BigQuery. 


 
![image](https://user-images.githubusercontent.com/17286177/207627105-71627d5e-5cad-4f98-ae42-cc58276ee4cc.png)

![image](https://user-images.githubusercontent.com/17286177/207627176-abae0243-966c-458f-a696-1fd338489254.png)



You can also see in this screenshot that the "Files" section is not red:

![image](https://user-images.githubusercontent.com/17286177/207632027-f3151d16-d51c-496f-91c6-f5de459a3965.png)

**Reply by briangow** [reply]
Did you also follow these instructions:https://mimic.mit.edu/docs/gettingstarted/cloud/ ? Do you have a gmail on your PhysioNet account: https://physionet.org/settings/emails/? What happens when you click "Request access using Google BigQuery" in your last screenshot?

**Reply by rajsinghUSA** [reply]
Yes, I already have the gmail linked in my profile.

![image](https://user-images.githubusercontent.com/17286177/207630952-bf1d3bca-5204-440e-bea0-a118bec95890.png)

Clicking on "Request access using Google BigQuery" shows this:

![image](https://user-images.githubusercontent.com/17286177/207631709-7feb9204-4ca8-4bdd-ac29-86162aca079f.png)

**Reply by rajsinghUSA** [reply]
In the PhysioNet GCP BigQuery access email, it says:
"If you cannot add the project "physionet-data", please access it through this link: https://console.cloud.google.com/bigquery?project=physionet-data&page=dataset".

Going to that link, I see some MIMIC3 data available:
![image](https://user-images.githubusercontent.com/17286177/207641540-ef01b5f6-8df2-4ad7-b397-b68d18d391c6.png)

**Comment by briangow** [comment]
Hmm, I'm not sure why you can't see `physionet-data` then. I assume you are signing into http://console.cloud.google.com/bigquery with the same gmail that was awarded access by clicking "Request access using Google BigQuery".

**Reply by rajsinghUSA** [reply]
Yes, same google account. And you can see it's letting me have some access to MIMIC3 via https://console.cloud.google.com/bigquery?project=physionet-data&page=dataset

**Comment by briangow** [comment, marked as answer]
@rajsinghUSA , my apologies, after speaking with my colleague it appears that we missed a step with the cloud setup when releasing v2.1 of MIMIC. Could you please try to +ADD DATA and search for `physionet-data` again.

**Reply by rajsinghUSA** [reply]
Great! I see there is now mimiciv available when going to https://console.cloud.google.com/bigquery?project=physionet-data&page=dataset :

![image](https://user-images.githubusercontent.com/17286177/207650387-4613b5b9-3516-4863-b148-fb010d659f15.png)

And if choosing `Star a project by name`, then "physionet-data" can be entered.

![image](https://user-images.githubusercontent.com/17286177/207651711-2f3e53d1-f81d-46c6-bb2c-ae3865002f70.png)

Thank you!

**Reply by briangow** [reply]
Great, thanks for your patience and feedback!

**Reply by mjbroerman** [reply]
Hi! I think the maintainers may have a similar situation for the "derived" datasets. Also, I used the method above to find `physionet-data` and star it, but the search function did not work for it or the derived datasets.

<img width="528" alt="image" src="https://user-images.githubusercontent.com/16464567/211651312-95fb6984-714a-4a8b-8329-0818a17b42cf.png">

I tried this url:

https://console.cloud.google.com/bigquery?project=physionet-data.mimiciii_derived&page=dataset

And it displays

<img width="940" alt="image" src="https://user-images.githubusercontent.com/16464567/211651818-c635feb1-7af8-4ef8-afc6-db8d5838b08f.png">

However, you cannot star it as the resource does not seem to exist. 

Am I correct that accessing this *derived* resource is the best way to run the queries for ![SOFA](https://github.com/MIT-LCP/mimic-code/blob/main/mimic-iii/concepts/severityscores/sofa.sql) and other severity scores?

Thanks for maintain this wonderful resource!

**Reply by alistairewj** [reply]
Yes you're correct. You need to separately be provided access to MIMIC-III in order to view the mimiciii_derived table however. Can you confirm you've been approved for access to MIMIC-III, successfully requested access to MIMIC-III on BigQuery, but still can't see the mimiciii_derived dataset?

**Reply by RanElgedawy** [reply]
I am trying to access the MIMIC-IV-note data from a jupyter notebook. I keep getting a 403 error: 403 Access Denied: Table physionet-data:mimiciv_note.discharge: User does not have permission to query table physionet-data:mimiciv_note.discharge, or perhaps it does not exist in location US.
![BigQuery](https://github.com/MIT-LCP/mimic-code/assets/95228400/9a81f007-3a3f-4ad4-be28-36b92d28c295)
![BigQuery2](https://github.com/MIT-LCP/mimic-code/assets/95228400/b9674f1c-6d6b-487a-a856-3e962c90035f)


Would some one please tell me what I might be doing wrong here?  @alistairewj @briangow

**Comment by tompollard** [comment]
The process for pinning a project on BigQuery has recently changed. See: https://github.com/MIT-LCP/mimic-code/discussions/1453 for a step-by-step description of the new process.

**Comment by alistairewj** [comment]
It's definitely going to be an issue with the credentials. It's more
typical to do gcloud auth with your Google account and let the default
credentials be imported by python. Might be worth trying that and not
manually setting credentials with a JSON file.
